activate

```diff
@@ -71,25 +82,45 @@ serial="${serial%"${serial##*[![:space:]]}"}"
 # detect live system
 live="false"
 if grep -qw "endless.live" /proc/cmdline; then
-    image="true"
+    live="true"
+fi
+
+
+# We don't want dummy serials sent
+if [ "${serial}" == "To be filled by O.E.M." ]; then
+  unset serial
 fi
 
-# format as JSON
-vars="image release vendor product serial live"
-export ${vars}
-json="$(python -c 'import json, os, sys; print json.dumps(dict((v,os.environ[v]) for v in sys.argv[1:]))' ${vars})"
+# Export vars for python
+export image release vendor product serial live
+
+# Convert to json
+json=$(python << END
+import json, os, sys
+
+activation={}
+for activation_var in ["image", "release", "vendor", "product", "serial"]:
+    if activation_var in os.environ:
+        activation[activation_var] = os.environ[activation_var]
+
+activation["live"] = os.environ["live"] == "true"
+
+print json.dumps(activation)
+END
+)
+
+echo "Sending: ${json}"
 
 # report in
 if ! out=$(wget -O - -q --method=PUT --body-data="${json}" "${SCRIPT}") ; then
```

Comment:
@ramcq This isn't going to work - wget keeps sending `<json_data>=` to the server so it fails json validation and curl isn't installed by default. I think I'll try rewriting this thing in Python from scratch.